### DISCUSSION TITLE
Mod setting for Synesthesia to base the color on the timing of the next object

### DISCUSSION THREAD

**Discussion by monochrome22** [ppy/osu#23985] [Ideas]
Seems difficult to discern the timings with the way Synesthesia currently works. Maybe after playing with it for a long time you would be able to make sense of stuff like this:
![what](https://github.com/ppy/osu/assets/122969285/cf8eb836-1134-44b9-a478-576b3311bcb2)

So it might be useful to add an option to base the combo color on the timings (beat snaps) between objects instead, like this:

![colors5](https://github.com/ppy/osu/assets/122969285/09a75a27-5730-4a96-bdd6-012e0ebe1022)
![colors4](https://github.com/ppy/osu/assets/122969285/bb160c8a-10af-48da-9e2d-821dce0325f1)

(illustrative purpose only, colors and numbers not accurate)

Essentially it shows you the timing of the next object like this.

**Comment by Paladynee** [comment]
If I understood right, your implementation of the coloring does not depend on the next object, but rather, the previous object. See image 1, the 4th object from the left, colored purple. If it were colored according to the next object, it would've been white. 

The group of 5 circles 1/4 of a beat away from eachother, correspond to a physical movement of 5 taps in quick succession. Therefore the first object in the group of 5 circles (the first 4th-8th objects in image 1) should've been colored purple. This is exactly what my suggestion #24342 does.

**Reply by monochrome22** [reply]
What I mean is, you would know the timing of the circle you're about to press. In my first example, if it's purple, you would know it's 1/2 from the press you just did, if it's white, you would know it's 1/4 from the press you just did. Saying if it's previous object or next object just seems like arguing semantics; it can be either depending on how you look at it.
The images should be relied on when understanding my proposal, not any text.

**Reply by Paladynee** [reply]
> The images should be relied on when understanding my proposal, not any text.

Then your proposal is about the `Previous hitobject based coloring` not `Next hitobject based coloring`.

**Reply by monochrome22** [reply]
When I said "base the color on the timing of the next object", I meant the color shows you the what the timing of the object you're about to press is (next object), do you understand?
Thinking about it like that, my phrasing makes sense. Like I said, whether it's next or previous just depends on how you think about it, it's arguing semantics.

**Reply by Paladynee** [reply]
Yeah, we just confused 2 perspectives. Your perspective was centered on the player and my perspective was centered on the hitobjects.
